src/overlays/actors/ovl_Tools/ac_tools.c

```diff
@@ -17,24 +18,143 @@ ActorProfile Tools_Profile = {
     /* */ sizeof(Tools),
     /* */ aTOL_actor_ct,
     /* */ aTOL_actor_dt,
-    /* */ (ActorFunc)none_proc1,
-    /* */ (ActorFunc)none_proc1,
+    /* */ (void*)none_proc1,
+    /* */ (void*)none_proc1,
     /* */ NULL,
 };
-#endif
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Tools/ac_tools/aTOL_actor_ct.s")
+ToolClip aTOL_clip;
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Tools/ac_tools/aTOL_actor_dt.s")
+void aTOL_init_clip_area(Game_Play* game_play);
+void aTOL_free_clip_area(void);
+void aTOL_secure_pl_umbrella_bank_area(Game_Play* game_play);
+s32 aTOL_check_data_bank(ObjectExchangeBank* objectExchangeBank, ToolName toolName, ToolActor* toolActor,
+                         s16 objectIndex);
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Tools/ac_tools/func_809657EC_jp.s")
+void aTOL_actor_ct(Actor* thisx UNUSED, Game_Play* game_play) {
+    aTOL_init_clip_area(game_play);
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Tools/ac_tools/func_8096595C_jp.s")
+void aTOL_actor_dt(Actor* thisx UNUSED, Game_Play* game_play UNUSED) {
+    aTOL_free_clip_area();
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Tools/ac_tools/func_80965A4C_jp.s")
+s32 aTOL_check_data_bank(ObjectExchangeBank* objectExchangeBank, ToolName toolName, ToolActor* toolActor,
+                         s16 objectIndex) {
+    s32 ret = -1;
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Tools/ac_tools/func_80965A70_jp.s")
+    if ((toolActor->actor.part == ACTOR_PART_PLAYER) && (toolName <= TOOL_UMBRELLA31) &&
+        (common_data.toolClip->umbrellaObjectIndex != -1)) {
+        ObjectStatus* objectStatus = &objectExchangeBank->status[common_data.toolClip->umbrellaObjectIndex];
+        Actor* temp_a0;
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Tools/ac_tools/func_80965ACC_jp.s")
+        if (((objectStatus->id >= 0) ? (objectStatus->id) : (-objectStatus->id)) != objectIndex) {
+            u32 objectSize = gObjectTable[objectIndex].vromEnd - gObjectTable[objectIndex].vromStart;
```

Comment:
as long as it matches
```suggestion
            size_t objectSize = gObjectTable[objectIndex].vromEnd - gObjectTable[objectIndex].vromStart;
```